desc/coils.py

```diff
@@ -1361,34 +1432,32 @@ def compute(
             )
         ]
 
-    def compute_magnetic_field(self, coords, params=None, basis="rpz", grid=None):
+    def compute_magnetic_field(self, grid, params=None, basis="rpz", source_grid=None):
         """Compute magnetic field at a set of points.
 
         Parameters
         ----------
-        coords : array-like shape(n,3) or Grid
-            coordinates to evaluate field at [R,phi,Z] or [x,y,z]
+        grid : array-like shape(n,3) or Grid
+            Nodes to evaluate field at in [R,phi,Z] or [X,Y,Z] coordinates.
         params : dict or array-like of dict, optional
-            parameters to pass to curves, either the same for all curves,
-            or one for each member
+            Parameters to pass to coils, either the same for all coils or one for each.
         basis : {"rpz", "xyz"}
-            basis for input coordinates and returned magnetic field
-        grid : Grid, int or None or array-like, optional
-            Grid used to discretize coil, either the same for all coils or one for each
-            member of the coilset. If an integer, uses that many equally spaced
-            points.
+            Basis for input coordinates and returned magnetic field.
+        source_grid : Grid, int or None or array-like, optional
+            Grid used to discretize coils. If an integer, uses that many equally spaced
+            points. Should NOT include endpoint at 2pi.
 
         Returns
         -------
         field : ndarray, shape(n,3)
             magnetic field at specified points, in either rpz or xyz coordinates
         """
         params = self._make_arraylike(params)
-        grid = self._make_arraylike(grid)
+        source_grid = self._make_arraylike(source_grid)
 
         B = 0
-        for coil, par, grd in zip(self.coils, params, grid):
-            B += coil.compute_magnetic_field(coords, par, basis, grd)
+        for coil, par, grd in zip(self.coils, params, source_grid):
```

Comment:
dont we need logic here as well to handle NFP and sym?

Reply:
Well this depends on what the purpose of the `MixedCoilSet` is. I am of the opinion that it should only be used to combine different types of coils at the highest level, so that objectives expecting a single "field thing" can compute the total magnetic field from a combination of different coils. In other words, I think every sub-member should be responsible for it's own symmetry (if applicable), and the `MixedCoilSet` simply sums all the magnetic fields together. In which case this method can be left the way it is. 

Otherwise it could get weird. Say for example we had a `MixedCoilSet(NFP=4, sym=True)` that was a combination of a `CoilSet` with the same symmetry, another `CoilSet` without symmetry, and a couple of individual coils. How should we decide to apply the symmetry for each sub-member? So I think it's best if this class doesn't know anything about stellarator or field period symmetry.

Reply:
Hm I see the point, I think I agree with Daniel. We should make this explicit in the description of this class though that it does NOT have any symmetry etc and so if it is used you should have to provide either coils covering the entire 2pi in Phi range, or sub coilsets which account for their own symmetries

Reply:
If we go with this plan, we might need to rethink having `MixedCoilSet` be a subclass of `CoilSet`, or at least add checks for `self.NFP = 1` and `self.sym = False` (since it will inherit those attributes from `CoilSet`)

Reply:
It can still be a subclass, but we just don't make NFP and sym args at init, so they're always set to 1/False. Should also make it clean in the docstring what this means etc.